gtsam_catkin/CMakeLists.txt

```
@@ -35,83 +37,77 @@ macro(get_run_deps)
+        CATKIN_DEPENDS
+            ${CATKIN_PACKAGE_DEPENDENCIES}
+            ${${PROJECT_NAME}_CATKIN_RUN_DEPENDS}
+        DEPENDS GTSAM
     )
-
-    # 2 - Check whether GTSAM is installed -------------------------------------------------------
+# 2 - Check whether GTSAM is installed -------------------------------------------------------
 elseif (GTSAM_FOUND)
-    message("${BoldMagenta}INFO: Found GTSAM installation.${ColourReset}")
+    message("${BoldMagenta}INFO: Found GTSAM installation at ${GTSAM_DIR}.${ColourReset}")
 
-    message("GTSAM include path: ${GTSAM_INCLUDE_DIR}")
-    message("GTSAM libs: ${GTSAM_LIBRARY_DIR}")
+    find_package(GTSAM_UNSTABLE REQUIRED)
 
     catkin_package(
-            CATKIN_DEPENDS ${CATKIN_PACKAGE_DEPENDENCIES}
-            DEPENDS GTSAM
-            INCLUDE_DIRS ${GTSAM_INCLUDE_DIR}
-            LIBRARIES
-            gtsam gtsam_unstable
+        INCLUDE_DIRS
+            ${GTSAM_INCLUDE_DIR}
+        LIBRARIES
+            gtsam
+            gtsam_unstable
+            metis-gtsam
             ${CS_PROJECT_LIBRARIES}
+        CATKIN_DEPENDS
+            ${CATKIN_PACKAGE_DEPENDENCIES}
+            ${${PROJECT_NAME}_CATKIN_RUN_DEPENDS}
+        DEPENDS GTSAM
     )
-
-    # Path differs whether installed from PPA or (locally) from source
-#    if (EXISTS "${GTSAM_DIR}/../../../../include/")
-#        set(GTSAM_INCLUDE_DIR_MAN "${GTSAM_DIR}/../../../../include/")
-#    else ()
-#        set(GTSAM_INCLUDE_DIR_MAN "${GTSAM_DIR}/../../../include/")
-#    endif ()
-
-    # 3 - Compile in catkin workspace -------------------------------------------------------------
+# 3 - Compile in catkin workspace -------------------------------------------------------------
 else ()
     message("${BoldMagenta}INFO: Neither variable GTSAM_DIR is set, nor could GTSAM be found, hence compiling in workspace.${ColourReset}")
 
-    # Newer version of CMake is required
-    # cmake_minimum_required(VERSION 3.18)
-
     # Catkinization of GTSAM
     include(ExternalProject)
     file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include)
 
     ExternalProject_Add(GTSAM
-            GIT_REPOSITORY "https://github.com/borglab/gtsam.git"
-            GIT_TAG "4.2a9"
-            GIT_PROGRESS "true"
-            CMAKE_CACHE_ARGS "-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true"
-            PREFIX "${CMAKE_BINARY_DIR}/gtsam_external"
-            SOURCE_DIR "${CMAKE_BINARY_DIR}/gtsam_external/src/gtsam"
-            BINARY_DIR "${CMAKE_BINARY_DIR}/gtsam_external/build"
-            CMAKE_ARGS
+        GIT_REPOSITORY "https://github.com/borglab/gtsam.git"
+        GIT_TAG "${GTSAM_TAG}"
+        GIT_PROGRESS "true"
+        PREFIX      "${CMAKE_SOURCE_DIR}"
+        TMP_DIR     "${CMAKE_BINARY_DIR}/gtsam/tmp"
+        BINARY_DIR  "${CMAKE_BINARY_DIR}/gtsam/build"
+        INSTALL_DIR "${CATKIN_DEVEL_PREFIX}"
+        CMAKE_CACHE_ARGS "-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true"
+        CMAKE_ARGS
             -DCMAKE_BUILD_TYPE=Release
```

```suggestion
            -DCMAKE_BUILD_TYPE=Release
            -DGTSAM_WITH_TBB=OFF
```
This is super helpful in terms of performance. Reduces peak cpu load by a factor of 3 for me